main.tf

```
@@ -94,18 +94,21 @@ resource "aws_lambda_function_event_invoke_config" "default" {
 
 resource "aws_lambda_function" "default" {
   provider                       = aws.lambda
-  function_name                  = var.name
   description                    = var.description
-  filename                       = local.filename
+  filename                       = var.s3_bucket == null ? local.filename : null
```

It would be better to leave `filename` as it was and let the provider fail when a plan runs if `s3_bucket` is set since we can't that kind of validation in a module.

---

Maybe we should do that before making a new release, but I'm fine with the current solution as well...

---

I would prefer to let the provider give an error that there is a clash instead of overriding values under the hood.

---

Fixed in #17